dart/lib/src/sentry_log_batcher.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:async';
+import 'sentry_envelope.dart';
+import 'sentry_options.dart';
+import 'protocol/sentry_log.dart';
+import 'package:meta/meta.dart';
+
+@internal
+class SentryLogBatcher {
+  SentryLogBatcher(this._options, {Duration? flushTimeout, int? maxBufferSize})
+      : _flushTimeout = flushTimeout ?? Duration(seconds: 5),
+        _maxBufferSize = maxBufferSize ?? 100;
+
+  final SentryOptions _options;
+  final Duration _flushTimeout;
+  final int _maxBufferSize;
+
+  final _logBuffer = <SentryLog>[];
+
+  Timer? _flushTimer;
+
+  void addLog(SentryLog log) {
+    _logBuffer.add(log);
+
+    _flushTimer?.cancel();
```

Comment:
canceling the timer here means that, in theory, if a log is added every 4 seconds, we will not flush them for several minutes
Is this the expectation? 
Just fyi, [Java sends every 5 seconds](https://github.com/getsentry/sentry-java/blob/ff3aa9c9755d65d99b1bc2788e6c0313df0665b1/sentry/src/main/java/io/sentry/logger/LoggerBatchProcessor.java#L62-L75), regardless of how many logs are there

Reply:
i guess it's fine as [per docs](https://develop.sentry.dev/sdk/telemetry/logs/#buffering), just ensuring it's wanted